Framework/Algorithms/test/ModeratorTzeroLinearTest.h

```diff
@@ -21,17 +18,21 @@ using Mantid::HistogramData::BinEdges;
 using Mantid::HistogramData::LinearGenerator;
 
 namespace {
-void AddToInstrument(MatrixWorkspace_sptr testWS,
+void addToInstrument(MatrixWorkspace_sptr testWS,
                      const bool &add_deltaE_mode = false,
                      const bool &add_t0_formula = false) {
   const double evalue(2.082); // energy corresponding to the first order Bragg
                               // peak in the analyzers
   if (add_deltaE_mode) {
     testWS->instrumentParameters().addString(
         testWS->getInstrument()->getComponentID(), "deltaE-mode", "indirect");
-    for (size_t ihist = 0; ihist < testWS->getNumberHistograms(); ++ihist)
-      testWS->instrumentParameters().addDouble(
-          testWS->getDetector(ihist)->getComponentID(), "Efixed", evalue);
+    const auto &spectrumInfo = testWS->mutableSpectrumInfo();
+    for (size_t ihist = 0; ihist < testWS->getNumberHistograms(); ++ihist) {
+      if (spectrumInfo.hasDetectors(ihist)) {
+        testWS->instrumentParameters().addDouble(&spectrumInfo.detector(ihist),
+                                                 "Efixed", evalue);
+      }
+    }
```

Comment:
I admit that this behavior is not nice, but we could not find another way in this transition period. This will be cleaned up when the Instrument-2.0 implementation has progressed further.